### PR TITLE
Sprint 4 - versification check

### PR DESCRIPTION
**Pull request by Ro-Bert-Web**
Added the verification phase to check whether chapter takes have the correct amount of verses.

- - -
This change is [Reviewable](https://reviewable.io/reviews/Bible-Translation-Tools/maui/61)

### CODE REVIEW

File: jvm/src/main/kotlin/org/bibletranslationtools/maui/jvm/mappers/FileVerifier.kt [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                        "$book only has ${chapterVerses.size} chapters, not $chapterNumber"
+                    )
+                } else {
+                    return VerifiedResult(FileStatus.PROCESSED, fileData.file)
+                }
+            }
+        }
+    }
+
+    private fun verseValid(fileData: FileDataItem): VerifiedResult {
```
Comment (AnonymousWalker) [inline jvm/src/main/kotlin/org/bibletranslationtools/maui/jvm/mappers/FileVerifier.kt]: I would rename this method to something like validateVerse() or isVerseValid()

File: jvm/src/main/kotlin/org/bibletranslationtools/maui/jvm/io/HtmlWriter.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package org.bibletranslationtools.maui.jvm.io
+
+import java.io.File
+
+class HtmlWriter {
+    fun write(filename:String, content:String) {
+        val file = File(filename)
+        file.printWriter().use {
+            it.println(content)
```
Comment (AnonymousWalker) [inline jvm/src/main/kotlin/org/bibletranslationtools/maui/jvm/io/HtmlWriter.kt:9]: are we going to leave it like this or add actual implementation later? Let's add a comment `// TODO` here explicitly.

File: jvm/src/main/kotlin/org/bibletranslationtools/maui/jvm/mappers/FileVerifier.kt [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import org.bibletranslationtools.maui.common.data.Grouping
+import org.bibletranslationtools.maui.common.data.VerifiedResult
+import org.bibletranslationtools.maui.common.io.Versification
+import org.bibletranslationtools.maui.jvm.ui.FileDataItem
+import org.wycliffeassociates.otter.common.audio.wav.CueChunk
+import org.wycliffeassociates.otter.common.audio.wav.WavFile
+import org.wycliffeassociates.otter.common.audio.wav.WavMetadata
+import tornadofx.isInt
+
+class FileVerifier(private val versification: SimpleObjectProperty<Versification>) {
```
Comment (AnonymousWalker) [inline jvm/src/main/kotlin/org/bibletranslationtools/maui/jvm/mappers/FileVerifier.kt]: I'm not sure why we need a javafx beans property here for versification. This introduces the unsafe non-null cast `!!` you have below.

**Review by AnonymousWalker — CHANGES_REQUESTED** [review]

File: jvm/src/main/kotlin/org/bibletranslationtools/maui/jvm/mappers/FileVerifier.kt [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        val book = fileData.book?.uppercase()
+        val chapter = fileData.chapter
+        val cueChunk = CueChunk()
+        val wavMetadata = WavMetadata(listOf(cueChunk))
+        WavFile(fileData.file, wavMetadata)
+
+        versification.value[book]!!.let { chapterVerses ->
+            val chapterNumber = chapter!!.toInt()
+            val expected = chapterVerses[chapterNumber - 1]
+            val actual = cueChunk.cues.size
```
Comment (AnonymousWalker) [inline jvm/src/main/kotlin/org/bibletranslationtools/maui/jvm/mappers/FileVerifier.kt]: in kotlin, "actual" is a keyword, we should avoid using it for variable name.

File: jvm/src/main/kotlin/org/bibletranslationtools/maui/jvm/io/HtmlWriter.kt (hunk 2 above)
Comment (Ro-Bert-Web) [reply jvm/src/main/kotlin/org/bibletranslationtools/maui/jvm/io/HtmlWriter.kt:9]: I don't know what you mean by "leave it like this" and insinuating that this isn't "actual implementation"

Comment (AnonymousWalker) [reply jvm/src/main/kotlin/org/bibletranslationtools/maui/jvm/io/HtmlWriter.kt:9]: Oh, I overlooked the writer part, thanks for the notice

**Review by AnonymousWalker — APPROVED** [review]

**Pull request merged**